lib/tests/streamlit/config_test.py

```diff
@@ -390,9 +390,14 @@ def test_update_config_with_toml(self):
         self.assertEqual(False, config.get_option("client.caching"))
 
     def test_set_option(self):
-        with pytest.raises(AssertionError) as e:
+        with self.assertLogs(logger="streamlit.config", level="WARNING") as cm:
             config._set_option("not.defined", "no.value", "test")
-        self.assertEqual(str(e.value), 'Key "not.defined" is not defined.')
+        self.assertEqual(
+            cm.output,
+            [
+                'WARNING:streamlit.config: "not.defined" is not a valid config option. If you previously had this config option set, it may have been removed.'
```

Comment:
Oh hm it's probably just the test that doesn't have the log prefix configured the same way as it is when actually running, so you get output that looks like the leading space isn't inserted automatically.

I'd just remove the leading space since otherwise you'll get a double-space when it actually runs, then change the test to not match against the exact error message (and probably just match against some substring of it) since I think checking the full message is overly restrictive anyway

Reply:
That's a good idea. thanks